From d52fa1f6d1b483f134b6bf32e7aff27492b401d5 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 7 Jul 2023 09:44:29 +0530
Subject: [PATCH 0451/1141] arm64: dts: st: enable usb-role-switch on
 stm32mp257f-ev1

Enable usb-role-switch on stm32mp257f-ev1, to support dual role USB2
with power delivery on Type-C connector.

Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Change-Id: I1f63d5bdff209c76454fe2733c3ed21da95a77ab
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 17517e8fc6ea..2cd098bfbf25 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -456,6 +456,16 @@ typec@35 {
 			reg = <0x35>;
 			interrupts-extended = <&intc_rpmsg 0>;
 			status = "okay";
+			connector {
+				compatible = "usb-c-connector";
+				label = "USB-C";
+
+				port {
+					typec_ep: endpoint {
+						remote-endpoint = <&dwc3_ep>;
+					};
+				};
+			};
 		};
 	};
 };
@@ -627,6 +637,11 @@ &usb3dr {
 
 	dwc3: usb@48300000 {
 		maximum-speed = "high-speed";
-		dr_mode = "peripheral";
+		usb-role-switch;
+		port {
+			dwc3_ep: endpoint {
+				remote-endpoint = <&typec_ep>;
+			};
+		};
 	};
 };
-- 
2.39.2

